File: server/php/src/Application/Domain/Enums/PaymentLinks/ValidFor.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Enums\PaymentLinks;

enum ValidFor: int {
    case ONE_DAY = 24;
    case TWO_DAYS = 48;
    case TWO_WEEKS = 336;
    case ONE_MONTH= 720;
}

File: server/ruby/app/business/domain/enums/payment_links/valid_for.rb
module Business
  module Domain
    module Enums
      module PaymentLinks
        module ValidFor
          ONE_DAY   = 24
          TWO_DAYS  = 48
          TWO_WEEKS = 336
          ONE_MONTH = 720

          ALL = [ONE_DAY, TWO_DAYS, TWO_WEEKS, ONE_MONTH].freeze

          module_function

          def from_raw(input)
            return nil if input.nil?
            ALL.find { |v| v == input.to_i }
          end

          def to_i(value)
            value
          end
        end
      end
    end
  end
end
